### DISCUSSION TITLE
new whisperX feature [research]

### DISCUSSION THREAD

**Discussion by m-bain** [m-bain/whisperX#963] [Polls]
Please vote which is most useful to you. Also let me know if any feature is really important for a big usecase, e.g. your business, or downstream research project

ctx: i have a few weeks until i start new job so have some free time. To a prosperous 2025 🫶

**Comment by sulutian** [comment]
Currently, when identifying movies, some clips are not transcribed. Can this be fixed?

**Reply by Barabazs** [reply]
Are you using subtitle edit?

**Reply by sulutian** [reply]
I put the subtitle file into SubtitleEdit and found that it has less content compared to the outputs from faster-whisper and whisper.

**Reply by Barabazs** [reply]
It doesn't seem like your issue is caused by whisperX though?

**Reply by sulutian** [reply]
I'm not entirely sure, but I transcribed multiple videos using WhisperX, faster-whisper, and whisper, and found that WhisperX sometimes misses certain segments. I tried adjusting the VAD threshold, which seemed to help a bit, but some segments still didn't have subtitles.

**Comment by tukipona** [comment]
I'd really appreciate it if the numerals in transcriptions could be more precisely aligned. (As mentioned in Limitations)

**Reply by m-bain** [reply]
did you try --supress_numerals flag?

**Reply by 7k50** [reply]
I agree strongly that numerals alignment, must be a highly desirable feature.

Here is someone's pull request proposing a solution: https://github.com/m-bain/whisperX/pull/986
I have tried it, at first glance it seems indeed to align numerals.



The lack of numerals alignment is a problem, since I parse `.srt` outputs through `highlight_words=True` with post-processing logic to reconstitute timestamped words into blocks of text, with timestamp URLs, etc. Trying to create custom logic around this and various edge cases, has been challenging. I would rather prefer rough timestamps on numerals in WhisperX, than none at all.

`--suppress_numerals` with a post-processing text normalizer, could have worked, if it didn't seem to give so peculiar results, e.g. with `text2num`:
> Unemployment now stands at seven point two million, up three hundred and fifty two thousand.
> Unemployment now stands at 7.2000000, up 300 and 52000.

**Comment by Infinitay** [comment]
If you plan on improving speaker identification/fingerprinting as well as diarization, would you consider the possibility of speech separation too? Although I suppose that is a little out of the domain of this library as this is more tailored towards transcription. 

eg. https://oxfordwaveresearch.com/products/cleaver/

**Comment by BradKML** [comment]
Multi-lingual (as in as many languages as Qwen 3 is willing to cover incl. Cantonese) https://qwen3.org/

**Comment by EXDjosemacedo** [comment]
You should add an option to diarize on left and right channels, with more accuracy than what is right now. Right now I'm splitting the audios, transcribing, then aligning and merging them. They misalign; this should be very straightforward
